src/main/kotlin/com/makarytskyi/rentcar/service/impl/OrderServiceImpl.kt

```diff
@@ -37,18 +38,20 @@ internal class OrderServiceImpl(
 
     override fun create(createOrderRequest: CreateOrderRequest): Mono<OrderResponse> {
         return createOrderRequest.toMono()
+            .doOnNext { validateDates(it.from, it.to) }
             .flatMap {
-                validateDates(it.from, it.to)
-                validateUserExists(it.userId)
+                Mono.zip(
+                    Mono.defer { validateUserExists(createOrderRequest.userId) }.subscribeOn(Schedulers.parallel()),
```

Comment:
why do you use a parallel scheduler?

Reply:
Yes, it is better to use boundedElasic, because there is I/O calls in validation functions.